test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -685,6 +690,25 @@
   "Support fact command versions"
   [:v4])
 
+(defn approx-facts-query []
+  "Get all fact names and their values converting values to text.
+   Returns a vector of maps following the format below:
+   [{:certname \"blarg\" :name \"fact1\" :value \"1\"}]"
+  (query-to-vec
+   "SELECT fp.path as name,
+           COALESCE(fv.value_string,
+                    cast(fv.value_integer as text),
+                    cast(fv.value_boolean as text),
+                    cast(fv.value_float as text),
+                    '') as value,
+           fs.certname
+    FROM factsets fs
+         INNER JOIN facts as f on fs.id = f.factset_id
+         INNER JOIN fact_values as fv on f.fact_value_id = fv.id
+         INNER JOIN fact_paths as fp on f.fact_path_id = fp.id
+    WHERE fp.depth = 0
+    ORDER BY name ASC"))
```

Comment:
We don't have to do this right now, but if you're inclined, since we're reindenting anyway, we could convert this to one of the reindentation-friendly line-oriented styles.  As one way:
```
(->> ["select fp.path as name,"
      "       coalesce(fv.value_string,"
      ...]
     (apply str)
     query-to-vec)
```